src/core/Form/FilterInput/FilterInput.test.tsx

```diff
@@ -0,0 +1,199 @@
+
+const TestFilterInput = (
+  <FilterInput
+    labelText="Label"
+    items={tools}
+    onFiltering={(filtered) => console.log(filtered)}
+    filterRule={filter}
+  />
+);
+
+test('should not have basic accessibility issues', async () =>
+  axeTest(TestFilterInput));
+
+test('snapshot matches', () => {
+  const { container } = render(TestFilterInput);
+  expect(container.firstChild).toMatchSnapshot();
+});
+
+describe('props', () => {
+  describe('className', () => {
+    it('has the given custom className', () => {
+      const { container } = render(
+        <FilterInput
+          className="custom-style"
+          labelText="Label"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      expect(container.firstChild).toHaveClass('custom-style');
+    });
+  });
+
+  describe('status', () => {
+    it('has the className when in error state', () => {
+      const { container } = render(
+        <FilterInput
+          className="custom-style"
+          labelText="Label"
+          status="error"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      expect(container.firstChild).toHaveClass('fi-filter-input--error');
+    });
+  });
+
+  describe('statusText', () => {
+    it('has the status text element', () => {
+      const { getByText } = render(
+        <FilterInput
+          statusText="Example status text"
+          labelText="Label"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      const statusText = getByText('Example status text');
+      expect(statusText).toHaveClass('fi-status-text');
+    });
+
+    it('will be added to input aria-describedby', () => {
+      const { getByRole } = render(
+        <FilterInput
+          statusText="Example status text"
+          id="123"
+          labelText="Label"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      expect(getByRole('textbox')).toHaveAttribute(
+        'aria-describedby',
+        '123-statusText',
+      );
+    });
+  });
+
+  describe('disabled', () => {
+    it('has disabled attribute and classname', () => {
+      const { container, getByRole } = render(
+        <FilterInput
+          disabled={true}
+          labelText="Label"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      expect(container.firstChild).toHaveClass('fi-filter-input--disabled');
+
+      const inputField = getByRole('textbox') as HTMLInputElement;
+      expect(inputField).toHaveAttribute('disabled');
+    });
+  });
+
+  describe('labelText', () => {
+    it('should be found ', () => {
```

Comment:
should there be a test for labelAlign as well? Maybe not necessary, but just thinking.

Reply:
Yes, there should be.

Reply:
Test is now added for `labelAlign` also.